specutils/spectra/spectrum1d.py

```diff
@@ -95,13 +101,29 @@ def __init__(self, flux=None, spectral_axis=None, wcs=None,
         # parse a given wcs. This is put into a GWCS object to
         # then be used behind-the-scenes for all specutils operations.
         if spectral_axis is not None:
-            # Ensure that the spectral axis is an astropy quantity
+            # Ensure that the spectral axis is an astropy Quantity
             if not isinstance(spectral_axis, u.Quantity):
```

Comment:
We're going to need to check if the instance might also be a `SpectralCoord` (otherwise we'll bail here and never continue on).

```suggestion
            if not isinstance(spectral_axis, (u.Quantity, SpectralCoord)):
```

Reply:
Won't isinstance([a SpectralCoord], u.Quantity) return True, since SpectralCoord is a subclass of Quantity? Seemed to work when I was testing, and thus I thought this was cleaner. Plus (in regards to your next comment) we can then check to see if the Quantity is specifically a SpectralCoord, and create one if it's not.

Reply:
> Won't isinstance([a SpectralCoord], u.Quantity) return True, since SpectralCoord is a subclass of Quantity?

Ah, yep. Neglected to think about the fact that we're inheriting from `Quantity`. You can ignore this suggestion, then.

> Plus (in regards to your next comment) we can then check to see if the Quantity is specifically a SpectralCoord, and create one if it's not.

We've already checked that the spectral axis is _either_ a `Quantity` _or_ a `SpectralCoord` a few lines up, so as soon as we defined it as one, then it isn't the other, so we know to create one.

Reply:
Implemented the rest of the suggestions, can this conversation be resolved since we're leaving as-is?

Reply:
Yep!